src/functional/getIn.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import coerceKeyPath from '../utils/coerceKeyPath';
+import { NOT_SET } from '../TrieUtils';
+import { get } from './get';
+
+export function getIn(collection, searchKeyPath, notSetValue) {
+  const keyPath = coerceKeyPath(searchKeyPath);
+  let i = 0;
+  while (i !== keyPath.length) {
+    // Intermediate null/undefined value along path
+    if (!collection) {
```

Comment:
Seems like this will cause problems with other falsy values. For example:

```js
let names = ["foo", "bar", ""];

names.map(name => getIn(name, "length"));
// will return
[3, 3, undefined]
```

Maybe it'd be better to use `collection == null` or be completely explicit with `collection === null || collection === undefined`.

Reply:
Thanks for the review! You're right that this is a strange outcome. Instead, I think getin() should treat non-data structure values as opaque, which will control the set of outcomes. I'll follow up